src/test/java/com/github/discvrseq/walkers/VcfToLuceneIndexerIntegrationTest.java

```diff
@@ -90,9 +98,56 @@ public void doBasicTest2() throws Exception {
         args.addRaw("-AN");
         args.addRaw("SampleList");
 
+        return args;
+    }
+
+    @Test
+    public void doBasicTest2() throws Exception {
+        File luceneOutDir = new File(getTmpDir(), "luceneOutDir");
+        if (luceneOutDir.exists())
+        {
+            FileUtils.deleteDirectory(luceneOutDir);
+        }
+
+        ArgumentsBuilder args = getBaseArgs(luceneOutDir);
         runCommandLine(args);
 
         File[] outputs = luceneOutDir.listFiles();
         Assert.assertEquals(5, outputs.length);
+
+        validateLuceneIndex(luceneOutDir);
+    }
+
+    @Test
+    public void doBasicTestWithMultithreading() throws Exception {
+        File luceneOutDir = new File(getTmpDir(), "luceneOutDir2");
+        if (luceneOutDir.exists())
+        {
+            FileUtils.deleteDirectory(luceneOutDir);
+        }
+
+        ArgumentsBuilder args = getBaseArgs(luceneOutDir);
+        args.addRaw("--threads");
+        args.addRaw("2");
+        runCommandLine(args);
+
+        File[] outputs = luceneOutDir.listFiles();
+        Assert.assertEquals(5, outputs.length);
+
+        validateLuceneIndex(luceneOutDir);
+    }
+
+    private void validateLuceneIndex(File indexPath) throws IOException, ParseException {
+        try (Directory indexDirectory = FSDirectory.open(indexPath.toPath());
+             IndexReader indexReader = DirectoryReader.open(indexDirectory)
+        ) {
+            IndexSearcher indexSearcher  = new IndexSearcher(indexReader);
+
+            QueryParser queryParser = new QueryParser("contig", new StandardAnalyzer());
```

Comment:
Per https://lucene.apache.org/core/2_9_4/queryparsersyntax.html, I think the following format for multi-valued queries works:

```
// Whatever fields we want to search go here
List<String> fieldLIst = new ArrayList<String>() {
            {
                add("contig");
                add("AF");
                add("pos");
            }
        };
        
        
MultiQueryParser contigPosQueryParser = new QueryParser(fieldList, "pos:>100000000 AND contig:=12")
MultiQueryParser contigAFQueryParser = new QueryParser(fieldList, "AF:>=0.5 AND contig:=12")
```

To match a string, if you need to, just write the string -- no need for a CONTAINS operator. To make a clause mandatory, prefix it with a +, e.g `AF:(+>=0.2 +<0.3)`, although you may only need to do that for string queries.

Reply:
what about the query that exercises variable samples? like "sites where sample1 is variable"

Reply:
And would you expect this to work? It's returning zero:
``` 
            MultiFieldQueryParser queryParser = new MultiFieldQueryParser(new String[]{"contig", "start", "POSITION"}, new StandardAnalyzer());

            TopDocs topDocs = indexSearcher.search(queryParser.parse("start:>3000 AND contig:=1"), 10);
            Assert.assertEquals(topDocs.totalHits.value, 9L);

            topDocs = indexSearcher.search(queryParser.parse("POSITION:diff_pos_same_ref_same_alt"), 10);
            Assert.assertEquals(topDocs.totalHits.value, 1L);
```